static/app/views/projectInstall/platform.tsx

```diff
@@ -145,39 +252,96 @@ export function ProjectInstallPlatform({location, params, route, router}: Props)
+        organization,
+        platform: recentCreatedProject.slug,
+        project_id: recentCreatedProject.id,
+      });
+
+      deleteRecentCreatedProject();
+    }
+
+    router.replace(
+      normalizeUrl(
+        `/organizations/${organization.slug}/projects/new/?referrer=getting-started&project=${recentCreatedProject.id}`
+      )
+    );
+  }, [
+    recentCreatedProject,
+    organization,
+    shallProjectBeDeleted,
+    deleteRecentCreatedProject,
+    router,
+  ]);
+
   useEffect(() => {
     // redirect if platform is not known.
     if (!platform.key || platform.key === 'other') {
       redirectToNeutralDocs();
     }
   }, [platform.key, redirectToNeutralDocs]);
 
+  if (!project) {
+    return null;
+  }
+
   if (!platform.id) {
     return <NotFound />;
   }
 
   const issueStreamLink = `/organizations/${organization.slug}/issues/`;
   const performanceOverviewLink = `/organizations/${organization.slug}/performance/`;
-  const gettingStartedLink = `/organizations/${organization.slug}/projects/${params.projectId}/getting-started/`;
   const showPerformancePrompt = performancePlatforms.includes(platform.id as PlatformKey);
   const isGettingStarted = window.location.href.indexOf('getting-started') > 0;
 
-  if (!project) {
-    return null;
-  }
-
   return (
     <Fragment>
       <StyledPageHeader>
         <h2>{t('Configure %(platform)s SDK', {platform: platform.name})}</h2>
         <ButtonBar gap={1}>
-          <Button
-            icon={<IconChevron direction="left" size="sm" />}
-            size="sm"
-            to={gettingStartedLink}
+          <Confirm
+            bypass={!shallProjectBeDeleted}
+            message={t(
+              "Hey, just a heads up - we haven't received any data for this SDK yet and by going back all changes will be discarded. Are you sure you want to head back?"
+            )}
+            priority="danger"
+            confirmText={t("Yes I'm sure")}
+            onConfirm={handleGoBack}
+            onClose={() => {
+              if (!recentCreatedProject) {
+                return;
+              }
+              trackAnalytics('project_creation.data_removal_modal_dismissed', {
+                organization,
+                platform: recentCreatedProject.slug,
+                project_id: recentCreatedProject.id,
+              });
+            }}
+            onRender={() => {
+              if (!recentCreatedProject) {
+                return;
+              }
+              trackAnalytics('project_creation.data_removal_modal_rendered', {
+                organization,
+                platform: recentCreatedProject.slug,
+                project_id: recentCreatedProject.id,
+              });
+            }}
           >
-            {t('Back')}
-          </Button>
+            <Button icon={<IconChevron direction="left" size="sm" />} size="sm">
+              {t('Back to Platform Selection')}
```

Comment:
since we don't display breadcrumbs in the header of the page, going back felt weird if users are landing on this page from other entry points, so I decided to rename it to "Back to Platform Selection"